fenix/app/src/main/java/org/mozilla/fenix/FenixOverlay.kt

```diff
@@ -0,0 +1,87 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material.Snackbar
+import androidx.compose.material.SnackbarHost
+import androidx.compose.material.SnackbarHostState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.tooling.preview.Preview
+import kotlinx.coroutines.launch
+import org.mozilla.fenix.datastore.debugDrawerEnabled
+import org.mozilla.fenix.datastore.debugSettings
+import org.mozilla.fenix.debugdrawer.ui.DebugDrawerFAB
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Overlay for presenting Fenix-wide content.
+ */
+@Composable
+fun FenixOverlay() {
+    val debugDataStore = LocalContext.current.debugSettings
+    val debugDrawerFabEnabled by debugDataStore.debugDrawerEnabled.collectAsState(initial = false)
+
+    FenixOverlayContent(
+        debugDrawerFabEnabled = debugDrawerFabEnabled,
+    )
+}
+
+@Composable
+private fun FenixOverlayContent(
+    debugDrawerFabEnabled: Boolean,
+) {
+    val snackbarState = remember { SnackbarHostState() }
+    val scope = rememberCoroutineScope()
+
+    Box(
+        modifier = Modifier.fillMaxSize(),
+    ) {
+        DebugDrawerFAB(
+            visible = debugDrawerFabEnabled,
+            onClick = {
+                scope.launch {
+                    snackbarState.showSnackbar("Show debug drawer")
+                }
+            },
+            modifier = Modifier.align(Alignment.Center),
+        )
```

Comment:
For now, I think it'll make sense to include just the debug fab in this wrapper composable until we have more debugging UI to overlay the app. In other words, when the drawer UI is added, this might move into an intermediary wrapper `DebugOverlay`, if it makes sense to do so.